client/src/pages/Lobby/Lobby.tsx

```diff
@@ -23,21 +28,89 @@ interface Prop {
 
 //Have a button to start game (logic doesn't need to be implemented yet)
 
-const Lobby: React.FC<Prop> = ({socket, lobbyId, userId }) => {
- const LeaveRoom = (e: FormEvent<HTMLButtonElement>) => {
+export const Lobby: React.FC<ILobbyProps> = ({socket, lobbyId, userId, setMembers, members }) => {
+ 
+ const [notifications, setNotifications] = useState<string[]>([]);
+ const navigate = useNavigate();
+ 
+ 
+ const leaveRoom = (e: MouseEvent<HTMLButtonElement>): void => {
   e.preventDefault();
   if (lobbyId && userId) {
-   socket.emit("leave_room", { lobbyId, userId });
+   socket.emit("leave_room", { lobbyId, userId }, (response: {success: boolean}) => {
+    if(response.success){
+        navigate("/");
+    }
+   });
   }
  };
 
+ const startGame = (e: MouseEvent<HTMLButtonElement>): void => {
+  e.preventDefault();
+  if (lobbyId && userId) {
+   socket.emit("start_game", { lobbyId, userId }, (response: {success: boolean}) => {
+    if(response.success){
+        navigate(`/game:${lobbyId}`);
+    }
+   });
+  }
+ };
+
+
+ useEffect(() => {
+  // should these be stored in handler functions
+    socket.on("player_joined", (lobbyMembers, user) => {
+        setMembers(lobbyMembers);
+        setNotifications((prevNotifications) => [...prevNotifications, `${user} has joined`])
+    })
+
+    socket.on("user_left", (lobbyMembers, user) => {
+        setMembers(lobbyMembers);
+        setNotifications((prevNotifications) => [...prevNotifications, `${user} has left`])
+    })
+
+    socket.on("user_disconnected", (lobbyMembers, user) => {
+      setMembers(lobbyMembers);
+      setNotifications((prevNotifications) => [...prevNotifications, `${user} has disconnected`])
+    })
+
+    socket.on("current_members", (lobbyMembers) => {
+      setMembers(lobbyMembers);
+    })
+
+    return () => {
+        socket.off("player_joined");
+        socket.off("user_left");
+        socket.off("user_disconnected");
+        socket.off("current_members");
+    }
+ }, [setMembers, setNotifications]);
```

Comment:
After reading the socket.io docs, it says that it's highly recommended to not register socket events in child components. So I believe these should be moved up to App.tsx and it should still work. But maybe still need a useEffect() in Lobby so that the component re-renders whenever a new user joins the lobby and thereby updating the members list.